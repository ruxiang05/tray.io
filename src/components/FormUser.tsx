import React from "react";
import FormInput from "./FormInput";
import SubmitButton from "./SubmitButton";
import { FormProps } from "./Form";

const FormUser = ({ changePage, handleChange, error }: FormProps) => (
  <div>
    <FormInput
      type="text"
      name="name"
      placeholder="Name"
      required
      handleChange={handleChange}
    />
    <FormInput
      type="text"
      name="role"
      placeholder="Role"
      handleChange={handleChange}
    />
    <FormInput
      type="email"
      name="email"
      placeholder="Email Address"
      required
      handleChange={handleChange}
    />
    <FormInput
      type="password"
      name="password"
      placeholder="Password"
      required
      handleChange={handleChange}
    />
    {error && <p className="form-error">{error}</p>}
    <SubmitButton changePage={changePage} disabled={error ? true : false} />
  </div>
);

export default FormUser;
