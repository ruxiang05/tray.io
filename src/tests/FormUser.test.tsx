import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import FormUser from "../components/FormUser";

describe("FormUser", () => {
  const mock = {
    changePage: jest.fn(),
    handleChange: jest.fn(),
    error: "undefined",
  };

  it("should render", () => {
    const tree = renderer.create(<FormUser {...mock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render the correct number of inputs", () => {
    const wrapper = shallow(<FormUser {...mock} />);
    expect(wrapper.find("FormInput").length).toEqual(4);
  });

  it("should render error if it exists", () => {
    const wrapper = shallow(<FormUser {...mock} />);
    expect(wrapper.find(".form-error")).toBeDefined();
  });

  it("should disable the submit button if there is an error", () => {
    const wrapper = shallow(<FormUser {...mock} />);
    expect(wrapper.find("SubmitButton").prop("disabled")).toBeDefined();
  });
});
