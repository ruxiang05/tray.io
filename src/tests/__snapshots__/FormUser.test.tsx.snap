// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUser should render 1`] = `
<div>
  <div
    className="form-input"
  >
    <label
      htmlFor="name"
    >
      name:
    </label>
    <span
      className="required-mark"
    >
       * 
    </span>
    <input
      name="name"
      onChange={[MockFunction]}
      placeholder="Name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="role"
    >
      role:
    </label>
    <input
      name="role"
      onChange={[MockFunction]}
      placeholder="Role"
      type="text"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="email"
    >
      email:
    </label>
    <span
      className="required-mark"
    >
       * 
    </span>
    <input
      name="email"
      onChange={[MockFunction]}
      placeholder="Email Address"
      required={true}
      type="email"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="password"
    >
      password:
    </label>
    <span
      className="required-mark"
    >
       * 
    </span>
    <input
      name="password"
      onChange={[MockFunction]}
      placeholder="Password"
      required={true}
      type="password"
    />
  </div>
  <p
    className="form-error"
  >
    undefined
  </p>
  <button
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;
